sphinx_rtd_theme/versions.html

```diff
@@ -1,34 +1,87 @@
-{% if READTHEDOCS %}
-{# Add rst-badge after rst-versions for small badge style. #}
-  <div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="{{ _('Versions') }}">
+<script type="text/javascript">
+ document.addEventListener("readthedocs-addons-data-ready", function(event) {
+     const config = event.detail.data();
+
+     const downloadsNameDisplay = {
+      pdf: "PDF",
+      epub: "EPUB",
+      htmlzip: "HTML",
+    };
+
+     // TODO: remove section if there is no content.
+     // TODO: add `rtd-current-item` to the selected one.
+     const flyout = `
+  <div class="rst-versions" data-toggle="rst-versions" role="note">
     <span class="rst-current-version" data-toggle="rst-current-version">
       <span class="fa fa-book"> Read the Docs</span>
-      v: {{ current_version }}
+      v: ${ config.versions.current.slug }
       <span class="fa fa-caret-down"></span>
     </span>
     <div class="rst-other-versions">
-      <dl>
-        <dt>{{ _('Versions') }}</dt>
-        {% for slug, url in versions %}
-          <dd><a href="{{ url }}">{{ slug }}</a></dd>
-        {% endfor %}
-      </dl>
-      <dl>
-        <dt>{{ _('Downloads') }}</dt>
-        {% for type, url in downloads %}
-          <dd><a href="{{ url }}">{{ type }}</a></dd>
-        {% endfor %}
-      </dl>
-      <dl>
-        {# Translators: The phrase "Read the Docs" is not translated #}
-        <dt>{{ _('On Read the Docs') }}</dt>
+      <div class="injected">
+        <dl>
+          <dt>{{ _('Languages') }}</dt>
+          ${ config.projects.translations.map(
+            (translation) => `
+            <dd><a href="${ translation.urls.documentation }">${ translation.slug }</a></dd>
+          `).join("\n")}
+        </dl>
+        <dl>
+          <dt>{{ _('Versions') }}</dt>
+          ${ config.versions.active.map(
+            (version) => `
+            <dd><a href="${ version.urls.documentation }">${ version.slug }</a></dd>
+          `).join("\n")}
+        </dl>
+        <dl>
+          <dt>{{ _('Downloads') }}</dt>
+          ${ Object.entries(config.versions.current.downloads).map(
+            ([name, url]) => `
+            <dd><a href="${ url }">${ downloadsNameDisplay[name] }</a></dd>
+          `).join("\n")}
+        </dl>
+        <dl>
+          <dt>{{ _('On Read the Docs') }}</dt>
+            <dd>
+              <a href="${ config.projects.current.urls.home }">{{ _('Project Home') }}</a>
+            </dd>
+            <dd>
+              <a href="${ config.projects.current.urls.builds }">{{ _('Builds') }}</a>
+            </dd>
+            <dd>
+              <a href="${ config.projects.current.urls.downloads }">{{ _('Downloads') }}</a>
+            </dd>
+        </dl>
+        <dl>
+          <dt>Search</dt>
           <dd>
-            <a href="//{{ PRODUCTION_DOMAIN }}/projects/{{ slug }}/?fromdocs={{ slug }}">{{ _('Project Home') }}</a>
+            <div style="padding: 6px;">
```

Comment:
Styles should go in sass/css instead of HTML, but is there a reason why this flyout menu is visually different from the standard flyout in the first place?

Reply:
I'm not sure why you are saying it's visually different from the standard flyout. What flyout are you comparing against?

In this PR I'm replicating the flyout that is injected as blob HTML from the backend and I don't want to make any style/html structure change compared to that one. This is to keep as much as we can backward compatibility at this step.

I plan to make CSS/HTML structure changes in a following iteration/release that performs breaking changes (e.g. integrate the versions/languages selector into the layout itself as you have mentioned in the past).

For now, I want to keep things simple and try to make the upgrade as smooth as possible (_if_ possible, 😄 )

Reply:
I'm comparing the flyout in this PR to the flyout you get without this PR. I was wondering why we needed the extra `padding` here.

But, looking at the standard flyout, it seems we used inline styles there too. Neither flyout should use this pattern, but it's okay that it matches for now.